prov/netdir/src/netdir_ep.c

```diff
@@ -850,5 +851,28 @@ static ssize_t ofi_nd_ep_cancel(fid_t fid, void *context)
 	return ofi_nd_cq_cancel(fid, context);
 }
 
-#endif /* _WIN32 */
+int ofi_nd_ep_getopt(struct fid* fid, int level, int optname,
+	void* optval, size_t* optlen)
+{
+	assert(fid->fclass == FI_CLASS_EP);
+	struct nd_ep* ep = container_of(fid, struct nd_ep, fid.fid);
+
+	assert(level == FI_OPT_ENDPOINT && optname == FI_OPT_CM_DATA_SIZE);
```

Comment:
This assert shouldn't be here.  The provider may not support other options, but there are valid values that an app can pass into this function.  The runtime check below is sufficient, so that we don't crash the app using a debug build.